alerting/src/main/kotlin/org/opensearch/alerting/TriggerService.kt

```diff
@@ -53,6 +55,37 @@ class TriggerService(val scriptService: ScriptService) {
         }
     }
 
+    // TODO: improve performance and support match all and match any
+    fun runDocLevelTrigger(
+        monitor: Monitor,
+        trigger: DocumentLevelTrigger,
+        ctx: DocumentLevelTriggerExecutionContext,
+        docsToQueries: Map<String, List<String>>,
+        queryIds: List<String>
+    ): DocumentLevelTriggerRunResult {
+        return try {
+            val triggeredDocs = mutableListOf<String>()
+
+            for (doc in docsToQueries.keys) {
+                val params = trigger.condition.params.toMutableMap()
+                for (queryId in queryIds) {
+                    params[queryId] = docsToQueries[doc]!!.contains(queryId)
+                }
+                val triggered = scriptService.compile(trigger.condition, TriggerScript.CONTEXT)
+                    .newInstance(params)
+                    .execute(ctx)
+                logger.info("trigger val: $triggered")
+                if (triggered) triggeredDocs.add(doc)
+            }
+
+            DocumentLevelTriggerRunResult(trigger.name, triggeredDocs, null)
+        } catch (e: Exception) {
```

Comment:
Instead of Document -> Query, can we do Query -> Document as number of documents will be fairly large compared to number of rules. Also it will simplify match all cases, since it will boil down to the problem of common ids across multiple lists. thoughts?

Reply:
This is still in discussion and we will have a path forward in a future PR.